packages/machine-extractor/src/extractAction.ts

```diff
@@ -0,0 +1,411 @@
+                };
+              } else {
+                assignment[prop.key.name] = {
+                  type: getLiteralType(prop.value),
+                  value: prop.value.value,
+                };
+              }
+            }
+          }
+        }
+      });
+
+      return assignment;
+    }
+
+    // assign(() => {})
+    // assign(function() {})
+    if (
+      t.isArrowFunctionExpression(assigner) ||
+      t.isFunctionExpression(assigner)
+    ) {
+      return {
+        inlineImplementation: {
+          type: 'expression',
+          value: fileContent.slice(assigner.start!, assigner.end!),
+        },
+      };
+    }
+  }
+
+  throw Error(`Unsupported assignment`);
+}
+
+export function extractRaisedEvent(
+  actionNode: ActionNode,
+  fileContent: string,
+): Record<
+  string,
+  {
+    type: JsonEntryType;
+    value: JsonValue;
+  }
+> {
+  const node = actionNode.node;
+  if (t.isCallExpression(node)) {
+    const arg = node.arguments[0];
+
+    // raise({type: 'event', ...props})
+    if (t.isObjectExpression(arg)) {
+      return extractEventObject(arg, fileContent);
+    }
+
+    // raise('event')
+    if (t.isStringLiteral(arg)) {
+      return { type: { type: 'string', value: arg.value } };
+    }
+  }
+
+  throw Error(`Unsupported raised event`);
+}
+
+export function extractLogExpression(
+  actionNode: ActionNode,
+  fileContent: string,
+): { type: JsonEntryType; value: JsonValue } {
+  const node = actionNode.node;
+  if (t.isCallExpression(node)) {
+    const arg = node.arguments[0];
+
+    // log('string')
+    if (t.isStringLiteral(arg)) {
+      return {
+        type: 'string',
+        value: arg.value,
+      };
+    }
+
+    // log((ctx, evt) => {})
+    if (t.isArrowFunctionExpression(arg) || t.isFunctionExpression(arg)) {
+      return {
+        type: 'expression',
+        value: fileContent.slice(node.start!, node.end!),
+      };
+    }
+  }
+
+  throw Error(`Unsupported log expression`);
+}
+
+export function extractStopProperties(
+  actionNode: ActionNode,
+  fileContent: string,
+): { type: JsonEntryType; value: JsonValue } {
+  const node = actionNode.node;
+  if (t.isCallExpression(node)) {
+    const arg = node.arguments[0];
+
+    if (t.isStringLiteral(arg)) {
+      return { type: 'string', value: arg.value };
+    }
+
+    if (t.isArrowFunctionExpression(arg) || t.isFunctionExpression(arg)) {
+      return {
+        type: 'expression',
+        value: fileContent.slice(arg.start!, arg.end!),
+      };
+    }
+  }
+
+  throw Error('Unsupported stop expression');
+}
+
+export function extractSendToProperties(
+  actionNode: ActionNode,
+  fileContent: string,
+): {
+  event:
+    | Record<
+        string,
+        {
+          type: JsonEntryType;
+          value: JsonValue;
+        }
+      >
+    | { type: 'expression'; value: string };
+  actorRef: {
+    type: JsonEntryType;
+    value: JsonValue;
+  };
```

Comment:
just reordered those things to match the order of arguments:
```suggestion
  actorRef: {
    type: JsonEntryType;
    value: JsonValue;
  };
  event:
    | Record<
        string,
        {
          type: JsonEntryType;
          value: JsonValue;
        }
      >
    | { type: 'expression'; value: string };
```